blaze-core/src/main/scala/org/http4s/blazecore/websocket/Http4sWSStage.scala

```diff
@@ -16,10 +17,11 @@ import org.http4s.websocket.WebSocketFrame._
 import scala.concurrent.ExecutionContext
 import scala.util.{Failure, Success}
 
-private[http4s] class Http4sWSStage[F[_]](
+private[http4s] class Http4sWSStage[F[_]] private (
     ws: WebSocket[F],
     sentClose: AtomicBoolean,
-    deadSignal: SignallingRef[F, Boolean]
+    deadSignal: SignallingRef[F, Boolean],
+    writeSemaphore: Semaphore[F]
```

Comment:
The change could be more minimal if we'd construct this semaphore in the constructor with a `.toIO.unsafeRunSync()`, an ugliness that was necessary anyway in blaze-server.

The constructor is privatized to avoid passing in a semaphore of more than one permit.

Reply:
That broke bincompat. It's a private[http4s], and we could get away with an exception, but those have burned us before.  Going with a more conservative approach.

Reply:
This broke bincompat.  It's `private[http4s]`, so we could get away with an exception, but I've been burned too many times by that.  1dc9664 is a more conservative approach.